From 137f2e5dc58ab63433c53bb0b3acf61c0b3b0628 Mon Sep 17 00:00:00 2001
From: Pete Skeggs <plskeggs@noeticdesign.com>
Date: Thu, 11 Jun 2026 12:16:01 -0700
Subject: [PATCH] Fix VrefO being always enabled

It was forced on for every mode of using the
Vref, regardless of which Vref to use:
- internal
- external Vref pin

Signed-off-by: Pete Skeggs <plskeggs@noeticdesign.com>
---
 drivers/regulator/regulator_nxp_vref.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/regulator/regulator_nxp_vref.c b/drivers/regulator/regulator_nxp_vref.c
index cb0bc5e11b4..1c8ea88cf8c 100644
--- a/drivers/regulator/regulator_nxp_vref.c
+++ b/drivers/regulator/regulator_nxp_vref.c
@@ -48,8 +48,9 @@ static int regulator_nxp_vref_enable(const struct device *dev)
 		;
 	}

-	/* Enable output buffer */
-	*csr |= VREF_CSR_BUF21EN_MASK;
+	/* Oresat: do not enable this! It outputs on P4_7, making that
+	   pin unusable for GPIO. */
+	*csr &= ~VREF_CSR_BUF21EN_MASK;

 	return 0;
 }
--
2.43.0

